### PR TITLE
Build local and remote tools depended on by command plugins before running the plugins

### PR DESCRIPTION
**Pull request by abertelrud**
When a command plugin has a dependency on an executable, it needs to be built or brought-up-to-date and included in the tool map.

### Modifications:

- extended the SwiftPackageTool logic to build these tools first
- other small fixes in the SwiftPackageTool logic for running plugins
- include them in the tools map sent to the command plugin
- extend a unit test to include this local and remote built tools

### CODE REVIEW

**Comment by abertelrud** [comment]
@swift-ci please smoke test

File: Sources/Commands/SwiftPackageTool.swift [hunk 1]
```diff
@@ -921,17 +922,25 @@ extension SwiftPackageTool {
             let plugin = matchingPlugins[0]
             swiftTool.observabilityScope.emit(info: "Running plugin \(plugin)")
             
-            // Find the targets (if any) specified by the user. We expect them in the root package.
+            // The proposal calls for applying plugins to regular targets.
+            func isEligible(target: ResolvedTarget) -> Bool {
+                return ![.plugin, .snippet, .systemModule].contains(target.type)
+            }
```
Comment (tomerd) [inline Sources/Commands/SwiftPackageTool.swift:928]: can this be made into an extension on ResolvedTarget?

File: Tests/CommandsTests/PackageToolTests.swift [hunk 2]
```diff
@@ -1170,27 +1170,37 @@ final class PackageToolTests: CommandsTestCase {
     func testCommandPlugin() throws {
```
Comment (tomerd) [inline Tests/CommandsTests/PackageToolTests.swift:1170]: is it possible to add a test for the binary dependency case?

**Review by tomerd — APPROVED** [review]
looks good, some small suggestions

File: Tests/CommandsTests/PackageToolTests.swift (hunk 2 above)
Comment (abertelrud) [reply Tests/CommandsTests/PackageToolTests.swift:1170]: Definitely, I actually have that in a later PR, but can bring that here.

File: Sources/Commands/SwiftPackageTool.swift (hunk 1 above)
Comment (abertelrud) [reply Sources/Commands/SwiftPackageTool.swift:928]: 👍

**Comment by abertelrud** [comment]
> looks good, some small suggestions

Thanks, implemented both.

**Comment by abertelrud** [comment]
@swift-ci please smoke test

**Comment by abertelrud** [comment]
This failure is in the updated unit test code, and only non-Darwin.  Will adjust it.

**Comment by abertelrud** [comment]
@swift-ci please smoke test

**Comment by abertelrud** [comment]
@tomerd are you fine with me landing this at this point?  I made the changes you suggested.

**Review by neonichu — APPROVED** [review]

**Pull request merged**